diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0679014..981349f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -311,21 +311,21 @@ include_directories(${SYSREPO_INCLUDE_DIRS})
 list(APPEND CMAKE_REQUIRED_INCLUDES ${SYSREPO_INCLUDE_DIRS})
 list(APPEND CMAKE_REQUIRED_LIBRARIES ${SYSREPO_LIBRARIES})
 
-# find sysrepoctl to be used for installation and tests
-if (NOT SYSREPOCTL_EXECUTABLE)
-    find_program(SYSREPOCTL_EXECUTABLE sysrepoctl)
-endif()
-if (NOT SYSREPOCTL_EXECUTABLE)
-    message(FATAL_ERROR "Unable to find sysrepoctl, set SYSREPOCTL_EXECUTABLE manually.")
-endif()
-
-# find sysrepocfg to be used for installation and tests
-if (NOT SYSREPOCFG_EXECUTABLE)
-    find_program(SYSREPOCFG_EXECUTABLE sysrepocfg)
-endif()
-if (NOT SYSREPOCFG_EXECUTABLE)
-    message(FATAL_ERROR "Unable to find sysrepocfg, set SYSREPOCFG_EXECUTABLE manually.")
-endif()
+# # find sysrepoctl to be used for installation and tests
+# if (NOT SYSREPOCTL_EXECUTABLE)
+#     find_program(SYSREPOCTL_EXECUTABLE sysrepoctl)
+# endif()
+# if (NOT SYSREPOCTL_EXECUTABLE)
+#     message(FATAL_ERROR "Unable to find sysrepoctl, set SYSREPOCTL_EXECUTABLE manually.")
+# endif()
+
+# # find sysrepocfg to be used for installation and tests
+# if (NOT SYSREPOCFG_EXECUTABLE)
+#     find_program(SYSREPOCFG_EXECUTABLE sysrepocfg)
+# endif()
+# if (NOT SYSREPOCFG_EXECUTABLE)
+#     message(FATAL_ERROR "Unable to find sysrepocfg, set SYSREPOCFG_EXECUTABLE manually.")
+# endif()
 
 # generate files
 configure_file("${PROJECT_SOURCE_DIR}/src/config.h.in" "${PROJECT_BINARY_DIR}/config.h" ESCAPE_QUOTES @ONLY)
